internal/adapter/provider/openrouter/codex_integration_test.go

```
@@ -0,0 +1,149 @@
+		Name:                 "test-openrouter",
+		Type:                 "openrouter",
+		SupportedClientTypes: []domain.ClientType{domain.ClientTypeClaude, domain.ClientTypeOpenAI},
+		Config: &domain.ProviderConfig{
+			OpenRouter: &domain.ProviderConfigOpenRouter{APIKey: "sk-or-test"},
+		},
+	})
+	if err != nil {
+		t.Fatalf("NewAdapter: %v", err)
+	}
+
+	req, _ := http.NewRequest(http.MethodPost, "http://localhost"+requestURI, nil)
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)